src/vos/vos_query.c

```diff
@@ -532,7 +532,8 @@ vos_obj_query_key(daos_handle_t coh, daos_unit_oid_t oid, uint32_t flags,
 
 	cont = vos_hdl2cont(coh);
 
-	vos_ts_set_add(query->qt_ts_set, cont->vc_ts_idx, NULL, 0);
+	rc = vos_ts_set_add(query->qt_ts_set, cont->vc_ts_idx, NULL, 0);
+	D_ASSERT(rc == 0);
```

Comment:
We cannot just raise the error up the stack?

Reply:
We could but this is what is done in all but 1 other call site. Every site just asserts 0.